test/Microsoft.AspNetCore.Mvc.FunctionalTests/ApiExplorerTest.cs

```diff
@@ -466,24 +531,77 @@ public ApiExplorerTest(MvcTestFixture<ApiExplorerWebSite.Startup> fixture)
 
             // Assert
             var description = Assert.Single(result);
-            Assert.Equal(type, description.ResponseType);
+            var responseType = Assert.Single(description.SupportedResponseTypes);
+            Assert.Equal(type, responseType.ResponseType);
+            Assert.Equal(200, responseType.StatusCode);
+            var responseFormat = Assert.Single(responseType.ResponseFormats);
+            Assert.Equal("application/json", responseFormat.MediaType);
         }
 
-        [Theory]
-        [InlineData("Controller", "ApiExplorerWebSite.Product")]
-        [InlineData("Action", "ApiExplorerWebSite.Customer")]
-        public async Task ApiExplorer_ResponseType_OverrideOnAction(string action, string type)
+        [Fact]
+        public async Task ApiExplorer_ResponseType_InheritingFromController()
         {
-            // Arrange & Act
+            // Arrange
+            var type = "ApiExplorerWebSite.Product";
+            var errorType = "ApiExplorerWebSite.ErrorInfo";
+
+            // Act
+            var response = await Client.GetAsync(
+                "http://localhost/ApiExplorerResponseTypeOverrideOnAction/Controller");
+
+            var body = await response.Content.ReadAsStringAsync();
+            var result = JsonConvert.DeserializeObject<List<ApiExplorerData>>(body);
+
+            // Assert
+            var description = Assert.Single(result);
+            Assert.Equal(2, description.SupportedResponseTypes.Count);
+            var responseType = Assert.Single(description.SupportedResponseTypes.Where(
+                responseFrmt => responseFrmt.ResponseType == type));
```

Comment:
Could you use collection asserts in a bunch of these places? Alternatively don't abbreviate them.